### PR TITLE
doc: new single-click setup onboarding

### PR DESCRIPTION
**Pull request by inge4pres**
Addresses the updates for 8.7

* Kibana single-click setup and instructions
* Migration guides (still to be written)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,21 +15,10 @@ at the end of this documentation for more information.
 
 Before setting up Universal Profiling, make sure you meet the following requirements:
 
-- An Elastic stack deployment on [Elastic Cloud](http://cloud.elastic.co) at version 8.5.0 or higher (you can either provision a new one or upgrade an existing one). Universal Profiling is currently only available on Elastic Cloud.
-- The workloads (to be profiled) must be running on Linux machines; with kernel version >= 4.15 
+- An Elastic stack deployment on [Elastic Cloud](http://cloud.elastic.co) at version 8.7.0 or higher (you can either provision a new one or upgrade an existing one). Universal Profiling is currently only available on Elastic Cloud.
+- The workloads (to be profiled) must be running on Linux machines, with x86_64 CPUs, with kernel version >= 4.15
```
Comment (rockdaboot) [inline README.md:19]:
```suggestion
- The workloads to be profiled must be running on Linux machines with x86_64 CPUs and kernel version >= 4.15.
```

File: README.md [hunk 2]
```diff
@@ -103,55 +73,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
+you will see a popup like the following.
 
-## Install the host-agent
+![image setup popup](./img/setup.png)
 
-The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile. It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 
+Click the button **Setup Universal Profiling** to complete this action. 
 
-To complete the basic setup of a host-agent on your Linux machine:
+Be aware that for the setup to be successful, you need to be logged into Kibana with a user with elevated privileges,
```
Comment (rockdaboot) [inline README.md:85]:
```suggestion
For the setup to be successful, you need to be logged into Kibana as a user with elevated privileges,
```

File: README.md [hunk 3]
```diff
@@ -103,55 +73,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
+typically the `elastic` user.
 
-1. Use `elastic-profiling` to print the host-agent installation and configuration instructions for various package formats.
-   The default package format is binary, but there are also instructions for DEB, RPM, Docker, and Kubernetes. The DEB and
-   RPM package formats contain systemd service definitions that ensure the host-agent stays running as a service.
+If you had already enabled Universal Profiling on the Cloud deployment and you are upgrading from a previous version,
```
Comment (rockdaboot) [inline README.md:88]:
```suggestion
If you already enabled Universal Profiling on the Cloud deployment and are going to upgrade from a previous version,
```

File: README.md [hunk 4]
```diff
@@ -103,55 +73,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
+see the [upgrade guide](./upgrade.md).
+Please note that, at this stage, upgrading **requires you to wipe all existing profiling data** and start from scratch.
```
Comment (rockdaboot) [inline README.md:90]:
```suggestion
**Warning**: At this stage, upgrading **requires you to wipe all existing profiling data**.
```

File: README.md [hunk 5]
```diff
@@ -103,55 +73,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
+You can always go to the instructions by clicking on the **Add data** button in the top-right corner of the page.
+
+Here are two examples of the provided instructions, for Kubernetes and Binary deployments.
+
+![image add data k8s](./img/add-data-kubernetes.png) ![image add data binary](./img/add-data-binary.png)
 
 **Host-agent configuration notes**
 
-- The printed configuration uses a `stable` version. This is good for testing environments. For production environmnets, we recommend using immutable versioned artifacts.
+- The instructions use a `stable` version. This is good for testing environments. For production environments,
```
Comment (rockdaboot) [inline README.md:107]: Not sure what this means =)

Comment (inge4pres) [reply README.md:106]: Do you mean that it's missing a piece of sentence or that it is not clear what users should do?

Comment (rockdaboot) [reply README.md:106]: The test says something about a "stable" HA release. How does a customer know what that means ? It implies that there are "unstable" versions available.
And then there are "immutable" versions of HA (for production).

For me this all doesn't make much sense. But how should a beta customer even feel ?

Comment (inge4pres) [reply README.md:106]: The points below explain how to get immutable versions

File: README.md [hunk 6]
```diff
@@ -156,7 +114,7 @@ To complete the basic setup of a host-agent on your Linux machine:
 [Elastic container library repository](https://container-library.elastic.co/r/observability/profiling-agent).
 
 - For Kubernetes deployments, the Helm chart version is already used to configure the same container image, unless
-overwritten with the `version` parameter in values.
+overwritten with the `version` parameter in te Helm values file.
```
Comment (rockdaboot) [inline README.md:117]:
```suggestion
overwritten with the `version` parameter in the Helm values file.
```

File: README.md [hunk 7]
```diff
@@ -98,55 +72,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
-   In this case, see [troubleshooting and support](#troubleshooting-and-support).
-1. After a few minutes, open Kibana and confirm that you can see stacktraces data coming from your host by
-   navigating to the **Threads** tab in **Observability > Universal Profiling > Stacktraces**. You should see a graph and a
-   list of processes.
-1. If everything is working correctly, you can now print more configurations to deploy the host-agents on your fleet.
+The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile.
+It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 
+
+To install a host-agent deployment and start gathering data, follow the instructions provided in the page.
+You can always go to the instructions by clicking on the **Add data** button in the top-right corner of the page.
```
Comment (mdbirnstiehl) [inline README.md:98]:
```suggestion
After clicking **Set up Universal Profiling** in the previous step, you'll see the instructions for installing the host-agent.
You can also find these instructions by clicking on the **Add data** button in the top-right corner of the page.
```

File: README.md [hunk 8]
```diff
@@ -176,12 +134,18 @@ You can also use the `-tags` flag to associate an arbitrary string with a specif
 Each tag must match `^[a-zA-Z0-9-:._]+$` regex and use `;` as a separator. Invalid tags are dropped
 and warnings issued on startup. 
 
-In Kibana, the KQL field `tags` can be used for filtering. For example:
+In Kibana, the KQL field `tags` can be used for filtering. For example, when running host-agent with the following:
```
Comment (mdbirnstiehl) [inline README.md:137]:
```suggestion
In Kibana, you can use the KQL field `tags` for filtering. For example, you can run the host-agent with the following:
```
Do we want to say what the result is that you'll get when you run the host-agent with the following?

File: README.md [hunk 9]
```diff
@@ -176,12 +134,18 @@ You can also use the `-tags` flag to associate an arbitrary string with a specif
...
 
 ```
 sudo pf-host-agent/pf-host-agent -project-id=1 -tags='cloud_region:us-central1;env:staging'
 ```
 
+in Kibana, you can filter profiling data from the host-agent with:
```
Comment (mdbirnstiehl) [inline README.md:143]:
```suggestion
in Kibana, you can filter profiling data from the host-agent with the following tag:
```

File: README.md (hunk 5 above)
Comment (mdbirnstiehl) [reply README.md:106]: Maybe we don't need to say stable? 

```suggestion
- The instructions in Kibana work well for testing environments. For production environments,
```

**Review by mdbirnstiehl — COMMENTED** [review]
A few comments here. I'm happy to discuss any questions or comments.

File: README.md [hunk 10]
```diff
@@ -98,55 +72,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
-1. Confirm that this is the first time setting up Universal Profiling in the terminal prompt.
-   If Universal Profiling is already installed, confirm with your cluster administrator that it's ok to erase all existing profiling data.
+If this is the first time enabling Universal Profiler on your Cloud deployment,
+you will see a popup like the following.
 
-## Install the host-agent
+![image setup popup](./img/setup.png)
 
-The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile. It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 
+Click the button **Setup Universal Profiling** to complete this action.
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply README.md:69]: Mhmhm I am pretty sure the button contains `Setup` so should we change it also in the UI?

File: README.md [hunk 11]
```diff
@@ -98,55 +72,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
 
-## Install the host-agent
+![image setup popup](./img/setup.png)
 
-The host-agent profiles your fleet, and you need to install and configure it on every machine that you want to profile. It needs elevated privileges in order to run (currently: `root` / `CAP_SYS_ADMIN`). 
+Click the button **Setup Universal Profiling** to complete this action. 
 
-To complete the basic setup of a host-agent on your Linux machine:
+For the setup to be successful, you need to be logged into Kibana as a user with elevated privileges,
+typically the `elastic` user.
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply README.md:72]: It's one of the pre-defined roles for Elasticsearch in Cloud, can be seen in the Security > Roles section

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply README.md:69]: Yep, sorry, that's what I meant to say.

File: README.md [hunk 12]
```diff
@@ -98,55 +72,39 @@ After you've enabled Universal Profiling in Kibana, it's located under **Observa
...
 
-1. Use `elastic-profiling` to print the host-agent installation and configuration instructions for various package formats.
-   The default package format is binary, but there are also instructions for DEB, RPM, Docker, and Kubernetes. The DEB and
-   RPM package formats contain systemd service definitions that ensure the host-agent stays running as a service.
+If you already enabled Universal Profiling on the Cloud deployment and are going to upgrade from a previous version,
+see the [upgrade guide](./upgrade.md).
+**Warning**: At this stage, upgrading **requires you to wipe all existing profiling data**.
+We recommend upgrading nonetheless, as we have added several improvements and new features in the latest version.
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply README.md:77]: Removing data _could_ be autoamted, but we were planning to document only the manual removal process.
See 09e4087

File: README.md (hunk 7 above)
Comment (inge4pres) [reply README.md:85]: Leaving this one pending as we need to clear the UI text first.

File: README.md (hunk 8 above)
Comment (inge4pres) [reply README.md:121]: Probably it's not evident: the result is described just below, it's that in Kibana you can add filters by tags.

https://github.com/elastic/universal-profiling-documentation/pull/29/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R127-R132

On the agent side, there's no difference in running with or without tags

File: README.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (inge4pres) [reply README.md:67]: Updated via https://github.com/elastic/kibana/pull/151828

**Review by rockdaboot — APPROVED** [review]
:+1: LGTM

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply README.md:74]: Sounds good. I updated the comment accordingly.

File: README.md [hunk 13]
```diff
@@ -176,12 +118,18 @@ You can also use the `-tags` flag to associate an arbitrary string with a specif
...
 and warnings issued on startup. 
 
-In Kibana, the KQL field `tags` can be used for filtering. For example:
+In Kibana, the KQL field `tags` can be used for filtering. For example, when running host-agent with the following:
 
 ```
 sudo pf-host-agent/pf-host-agent -project-id=1 -tags='cloud_region:us-central1;env:staging'
 ```
 
+in Kibana, you can filter profiling data from the host-agent with the following tag:
```
Comment (mdbirnstiehl) [inline README.md:127]:
```suggestion
You can then filter profiling data from the host-agent in Kibana with the following tag:
```

File: README.md (hunk 8 above)
Comment (mdbirnstiehl) [reply README.md:121]: Ok, I understand. I updated the suggestion.

**Review by mdbirnstiehl — APPROVED** [review]
I updated a couple of the suggestions after your explanations. With those updates, LGTM!

**Comment by inge4pres** [comment]
> I updated a couple of the suggestions after your explanations. With those updates, LGTM!

Thanks! Committed now.
There's a whole new section also

https://github.com/elastic/universal-profiling-documentation/blob/3606310e5d2840d2828ff1b5bc69c2d1df258c0c/upgrade.md

If you saw it already and all is well-written, you can ignore my comment.

**Comment by mdbirnstiehl** [comment]
@inge4pres Sorry, I missed that. I'll take a look today.

**Review by mdbirnstiehl — COMMENTED** [review]
Sorry, it's kind of a lot, but wanted to make sure this is consistent with the other docs we've put together. As always, let me know if you have any questions or comments.

File: upgrade.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+4. Setup Universal Profiling from scratch
+5. Start profiling data ingestion
+
+#### Perform a stack upgrade
+
+- Locate the deployment you wish to upgrade in [Cloud console](https://cloud.elastic.co)
+- Click the cog icon on the right of deployment's name to enter the deployment settings
+- Find **Deployment version** on the right of the page
+- Click on **Upgrade** and choose the desired version
+- Verify the process completes successfully
```
[the remark this answers is not in the document]
Comment (inge4pres) [reply upgrade.md:30]: Yeah this is the correct way 👍🏼

File: upgrade.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+Select all items, then click on **Manage indices**, then **Delete indices** in the menu.
+Confirm your intention by clicking **Delete indices** in the popup.
+
+Switch to the **Data Streams** tab, type `profiling-` in the search bar again.
+
+Select all items, then click on **Delete data streams**.
+Confirm your intention by clicking **Delete data streams** in the popup.
+
+You have now deleted all existing profiling data.
+
```
[the remark this answers is not in the document]
Comment (mdbirnstiehl) [reply upgrade.md:64]: My suggestion was that we could probably remove it as we wouldn't want the user to read that line and not continue to the next line thinking they didn't have anything else to do.

**Pull request merged**